### PR TITLE
Add about page

### PR DESCRIPTION
**Pull request by edfloreshz**
This PR implements the About page for COSMIC applications.

- Added `About` struct to store the application's information.
- Added `about` method in `Application` trait to provide information about the application.
- Added `about_view` in `Application` trait with default implementation to render the About page.
- Added optional dependency `open` to handle `on_press` events for URLs in the about page. 

image

### CODE REVIEW

File: src/app/mod.rs [hunk 1]
```diff
@@ -589,6 +600,102 @@ where
         panic!("no view for window {id:?}");
     }
 
+    #[cfg(feature = "desktop")]
+    /// Provides information about the application.
+    fn about(&self) -> Option<About> {
```
Comment (mmstick) [inline src/app/mod.rs:605]: This should return `Option<&About>` so we don't need to recreate it on each view update.

Comment (edfloreshz) [reply src/app/mod.rs]: Should be fixed, added an example as well.

**Review by mmstick — DISMISSED** [review]

**Comment by mmstick** [comment]
PR needs to be rebased, and then it should be good.

**Review by mmstick — APPROVED** [review]

**Review by mmstick — DISMISSED** [review]

**Comment by mmstick** [comment]
Needs to drop the change updating iced since that's causing a conflict

**Review by mmstick — APPROVED** [review]

**Pull request merged**